emr_cost_calculator.py

```diff
@@ -48,89 +43,131 @@ def validate_date(date_text):
         raise ValueError('Incorrect data format, should be YYYY-MM-DD')
 
 
-def retry_if_EmrResponseError(exception):
+def is_error_retriable(exception):
     """
     Use this function in order to back off only
-    on EmrResponse errors and not in other exceptions
+    if error is retriable
     """
-    return isinstance(exception, boto.exception.EmrResponseError)
+    # TODO verify if this is correct way to handle this. Haven't seen errors like this myself
+    try:
+        return exception.response['Error']['Code'].startswith("5")
+    except AttributeError:
+        return False
 
 
 class Ec2Instance:
-
-    def __init__(self, creation_ts, termination_ts, instance_price):
-        self.lifetime = self._get_lifetime(creation_ts, termination_ts)
-        self.cost = self.lifetime * instance_price
-
-    @staticmethod
-    def _parse_dates(creation_ts, termination_ts):
-        """
-        :param creation_ts: the creation time string
-        :param termination_ts: the termination time string
-        :return: the lifetime of a single instance in hours
-        """
-        date_format = '%Y-%m-%dT%H:%M:%S.%fZ'
-        creation_ts = \
-            time.mktime(time.strptime(creation_ts, date_format))
-        termination_ts = \
-            time.mktime(time.strptime(termination_ts, date_format))
-        return creation_ts, termination_ts
-
-    def _get_lifetime(self, creation_ts, termination_ts):
-        """
-        :param creation_ts: the creation time string
-        :param termination_ts: the termination time string
-        :return: the lifetime of a single instance in hours
-        """
-        (creation_ts, termination_ts) = \
-            Ec2Instance._parse_dates(creation_ts, termination_ts)
-        return math.ceil((termination_ts - creation_ts) / 3600)
+    def __init__(self, creation_ts, termination_ts, instance_type, market_type):
+        # creation_ts (EMR instance group parameter) correlates to EC2 instance start up time
+        self.creation_ts = creation_ts
+        self.termination_ts = termination_ts
+        self.instance_type = instance_type
+        self.market_type = market_type
 
 
 class InstanceGroup:
-
-    def __init__(self, group_id, instance_type, group_type):
+    def __init__(self, group_id, instance_type, market_type, group_type):
         self.group_id = group_id
         self.instance_type = instance_type
+        self.market_type = market_type
         self.group_type = group_type
-        self.price = 0
 
 
-class EmrCostCalculator:
+class Ec2EmrPricing:
+    def __init__(self, region):
```

Comment:
this method is too long imo, can you please split it in different "private" methods?
Also please provide a new public method on `Ec2EmrPricing` ex. `fetchPrices()` where all this should happen? Because is not ideal having so much logic inside a constructor.